toolchain/check/testdata/interop/cpp/no_prelude/cpp_diagnostics.carbon

```diff
@@ -450,6 +450,47 @@ library "[[@TEST_NAME]]";  // Trailing comment
 // CHECK:STDERR:
 import Cpp library "one_warning.h";
 
+// ============================================================================
+// Diagnostic accesses LangOptions
+// ============================================================================
+
+// --- lang_options.h
+
+template <>
```

Comment:
Would you mind adding a comment about what exactly causes it to touch LangOptions?